createWebNavigator.js

```diff
@@ -1,39 +1,38 @@
 import _ from 'underscore';
 import * as React from 'react';
-import {View, Pressable} from 'react-native';
+import {View, Dimensions} from 'react-native';
 import {useNavigationBuilder, createNavigatorFactory, StackRouter} from '@react-navigation/native';
+import RHPContainer from './components/RHPContainer';
+
+const isSmallScreenWidth = Dimensions.get('window').width <= 800;
+
 
 function CustomStackNavigator(props) {
     const {navigation, state, descriptors, NavigationContent} = useNavigationBuilder(StackRouter, {
       children: props.children,
       screenOptions: props.screenOptions,
       initialRouteName: props.initialRouteName,
     });
-
+    const isLHNVisible = (state.index === 0 && state.routes.length === 1) || !isSmallScreenWidth;
     const lastChatIndex = _.findLastIndex(state.routes, {name: 'Chat'});
     const lastRHPIndex = _.findLastIndex(state.routes, (route) => route.name !== 'Chat' && route.name !== 'LeftHandNav');
     const isRHPOnTopOfStack = lastRHPIndex === state.index;
     return (
         <NavigationContent>
             <View style={{flexDirection: 'row', flex: 1}}>
                 {state.routes.map((route, i) => {
-                    if (route.name === 'LeftHandNav') {
+                    if (isLHNVisible && i === 0) {
```

Comment:
With `i === 0` it's more like saying `route.name === 'LeftHandNav'`. So not sure if at all there is any improvement here.

Reply:
Can we pass an option to the screen/route so that the custom navigator can identify it as the LHN without checking against the `name`? P.S. we don't _need_ to take the feedback of the maintainer in this case if we don't want to and would rather rely on the name index or anything else.

Reply:
FWIW I really don't like using index here.